lib/engine/game/g_1860.rb

```diff
@@ -223,6 +255,125 @@ def close_other_companies!(company)
         @companies.reject { |c| c == company }.each(&:close!)
         @log << '-- Event: starting private companies close --'
       end
+
+      def biggest_train(corporation)
+        corporation.trains.map(&:distance).max || 0
+      end
+
+      def get_token_cities(corporation)
+        tokens = []
+        hexes.each do |hex|
+          hex.tile.cities.each do |city|
+            next unless city.tokened_by?(corporation)
+
+            tokens << city
+          end
+        end
+        tokens
+      end
+
+      def node_distance_walk(node, distance, node_distances: {}, corporation: nil, path_distances: {})
+        return if node_distances&.[](node) && node_distances[node] <= distance
```

Comment:
node_distancecs be nil?

if it's not you can do (node_distances[node] || 0) <= distance

Reply:
OK

Reply:
(node_distances[node] || 999) <= distance
Since I don't want to return if it isn't defined yet.